src/Nerdbank.Streams/NestedStream.cs

```diff
@@ -25,6 +25,11 @@ internal class NestedStream : Stream, IDisposableObservable
         /// <summary>
         /// The remaining bytes allowed to be read.
         /// </summary>
+        private long remainingBytes;
+
+        /// <summary>
+        /// The total length of the stream.
+        /// </summary>
         private long length;
```

Comment:
`length` should be `readonly` to avoid bugs like those that might exist if you forgot to change some of the code below that reassigns the field.